src/pudl/transform/ferc1.py

```diff
@@ -853,6 +853,533 @@ def strip_col_of_non_ints(self, col: pd.Series) -> pd.DataFrame:
+        df_labeled = (
+            df.pipe(label_header_rows)
+            .pipe(label_total_rows)
+            .pipe(label_notes_rows)
+            .drop(columns=["possible_header"])
+        )
+
+        return df_labeled
+
+    def map_header_fuel_types(
+        self, df: pd.DataFrame, show_unmapped_headers=False
+    ) -> pd.DataFrame:
+        """Apply the plant type indicated in the header row to the relevant rows.
+
+        This function groups the data by utility, year, and header and forward fills the
+        cleaned plant type based on that. As long as each utility year group that uses a
+        header for one plant type also uses headers for other plant types this will
+        work. I.e., if a utility's plant_name_ferc1 column looks like this: [STEAM,
+        coal_plant1, coal_plant2, wind_turbine1], this algorythem will think that wind
+        turbine is steam. Ideally (also usually) if they label one, they will label all.
+        The ideal version is:
+
+        [STEAM, coal_plant1, coal_plant2, WIND, wind_turbine]. Right now, this function
+        puts the new header plant type into a column called plant_type_2. This is so we
+        can compare against the current plant_type column for accuracy and validation
+        purposes.
+        """
+        logger.info(f"{self.table_id.value}: Mapping header fuels to relevant rows")
+
+        def expand_dict(dic):
+            """Change format of header_labels.
+
+            Right now, the header_labels dictionaries defined above follow this format:
+
+            {clean_name_A: [ bad_name_A1, bad_name_A2, ...], clean_name_B: [bad_name_B1,
+            bad_nameB2]}. This is convenient visually, but it makes more sense to format
+            it like this: {bad_name_A1: clean_name_A, bad_name_A2: clean_name_A,
+            bad_name_B1: clean_name_B, bad_name_B2: clean_name_B}. We could reformat the
+            dictionaries, but for now I created this function to do it where necessariy
+            in certain functions.
+            """
+            d = {}
+            for k, lst in dic.items():
+                for i in range(len(lst)):
+                    d[lst[i]] = k
+            return d
+
+        header_labels = {  # Add these to the PLANT_CATEGORIES param ideally
+            "hydroelectric": ["hydro", "hyrdo"],
+            "internal combustion": ["internal", "interal", "international combustion"],
+            "combustion turbine": ["combustion turbine"],
+            "combined cycle": ["combined cycle"],
+            "gas turbine": ["gas"],
+            "petroleum liquids": ["oil", "diesel", "diesal"],
+            "solar": ["solar", "photovoltaic"],
+            "wind": ["wind"],
+            "geothermal": ["geothermal"],
+            "waste": ["waste", "landfill"],
+            "steam": ["steam"],
+            "nuclear": ["nuclear"],
+            "fuel_cell": ["fuel cell"],
+            "other": ["other"],
+            "renewables": ["renewables"],
+        }
```

Comment:
Question for @zaneselvans or @cmgosnell. The headers I've found in the data pertain both to fuel_type and plant_type, and I'm not exactly sure how I should handle this. Here's the dictionary I was using in my old code (where the list keys are the different headings found in the data and the values are my attempt to standardize them). In looking at the `FUEL_CATEGORIES` and `PLANT_TYPE_CATEGORIES` in the params file, it seems like there's some overlap. I'm not sure whether I should attempt to

A) pick either `fuel_type` or `plant_type` and fit each of these headers into the relevant categories for either;

or B) try and update `plant_type` when there's plant type information and `fuel_type` when there's fuel type information.

"hydroelectric": ["hydro", "hyrdo"],
"internal combustion": ["internal", "interal", "international combustion"],
"combustion turbine": ["combustion turbine"],
"combined cycle": ["combined cycle"],
"gas turbine": ["gas"],
"petroleum liquids": ["oil", "diesel", "diesal"],
"solar": ["solar", "photovoltaic"],
"wind": ["wind"],
"geothermal": ["geothermal"],
"waste": ["waste", "landfill"],
"steam": ["steam"],
"nuclear": ["nuclear"],
"fuel_cell": ["fuel cell"],
"other": ["other"],
"renewables": ["renewables"],

Reply:
if values in these header rows are applicable for either `plant_type` or `fuel_type`... is there any reason why were wouldn't just run the same inputs through `categorize_strings` for the `plant_type` and `fuel_type`. There is lots of crossover in these types.

Reply:
That's a fair point!

Reply:
Added a hydro category to `PLANT_TYPE_CATEGORIES` and kept the `PLANT_TYPE_CATEGORIES_HYDRO` table for the hydro tables because it categories certain terms (like "conventional") differently. 

For the outliers, I'm just going to ignore them for now, though Christina suggested muting them.